### PR TITLE
#2249: Add perf test for raw MPI Send/Recv and ObjGroup's Send

### LINKED ISSUE #2249: Write benchmark to compare raw MPI send to objgroup send and collection send
**What Needs to be Done?**

We want to compare, for different payload sizes, a raw MPI send/recv with an objgroup send and a collection send of the same size.

### PR DESCRIPTION
**Pull request by JacobDomagala**
Fixes #2249 

`BuildStats` issue - https://github.com/DARMA-tasking/build-stats/issues/31

### CODE REVIEW

**Comment by JacobDomagala** [comment]
With the most recent commit:
### Raw MPI
Screenshot from 2024-04-25 15-03-34

### Objgroup send with payload wrapped in `shared_ptr`
Screenshot from 2024-04-25 15-03-53

### Objgroup send with payload as `const&`
Screenshot from 2024-04-25 15-05-30

File: tests/perf/send_cost.cc [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+  auto const prevNode = (thisNode - 1 + num_nodes_) % num_nodes_;
+  auto const nextNode = (thisNode + 1) % num_nodes_;
+  int data = thisNode;
+
+  for (auto size : payloadSizes) {
+    std::vector<int32_t> dataVec(size, data);
+
+    StartTimer(fmt::format("Payload size {}", size));
+
+    std::vector<int32_t> recvData(size, data);
```
Comment (JacobDomagala) [inline tests/perf/send_cost.cc:86]: This is "inside" timer block to match the allocation on vt's receiving side. I think this is fair.

**Comment by JacobDomagala** [comment]
### Raw MPI
image

### ObjGroup send
image

### Collection send
image

**Comment by JacobDomagala** [comment]
Next step would be to update `BuildStats' to plot the results

**Comment by lifflander** [comment]
@JacobDomagala This look good to me!

**Pull request merged**